packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -635,28 +632,39 @@ export class InputTimePicker
   }
 
   private delocalizeTimeStringToParts(
-    localizedTimeString: string,
+    value: string,
     fractionalSecondFormatToken?: "S" | "SS" | "SSS",
   ): DayjsTimeParts {
-    const ltsFormatString = this.localeConfig?.formats?.LTS;
-    const fractionalSecondTokenMatch = ltsFormatString.match(/ss\.*(S+)/g);
+    let localizedTimeString = value;
+    const effectiveHourFormat = isLocaleHourFormatOpposite(this.hourFormat, this.effectiveLocale)
+      ? getLocaleOppositeHourFormat(this.effectiveLocale)
+      : getLocaleHourFormat(this.effectiveLocale);
 
-    if (fractionalSecondFormatToken && this.shouldIncludeFractionalSeconds()) {
-      const secondFormatToken = `ss.${fractionalSecondFormatToken}`;
-      this.localeConfig.formats.LTS = fractionalSecondTokenMatch
-        ? ltsFormatString.replace(fractionalSecondTokenMatch[0], secondFormatToken)
-        : ltsFormatString.replace("ss", secondFormatToken);
-    } else if (fractionalSecondTokenMatch) {
-      this.localeConfig.formats.LTS = ltsFormatString.replace(fractionalSecondTokenMatch[0], "ss");
+    if (
+      localizedTwentyFourHourMeridiems.has(this.effectiveLocale) &&
+      effectiveHourFormat === "12"
+    ) {
+      const localizedAM = localizedTwentyFourHourMeridiems.get(this.effectiveLocale).am;
+      const localizedPM = localizedTwentyFourHourMeridiems.get(this.effectiveLocale).pm;
+      const localizedAMRegEx = new RegExp(String.raw`${localizedAM}`, "g");
+      const localizedPMRegEx = new RegExp(String.raw`${localizedPM}`, "g");
+      const meridiemFormatToken = getMeridiemFormatToken(this.effectiveLocale);
+      const caseAdjustedAMString =
```

Comment:
Is it possible to get the corresponding-cased token by looking at either the default or generated LT/LTS formats?

Reply:
> ...default or generated LT/LTS formats?

Are you referring to `dayjs`'s ones from each locale's config object?

Reply:
Yes.

Reply:
While that is possible, it can't be taken advantage of here because `dayjs` does not support storing more than one `LT` or `LTS` format string per locale, only the default string for the given locale is really allowed.  So for 24-hour default locales, the `a | A` meridiem format token would be absent from the config.

This branch of logic basically has to compute what the 12-hour format string should be for each locale specified in the `localizedTwentyFourHourMeridiems` Map, which are the supported locales that have unique localized meridiems that differ from the standard `am | pm | AM | PM` values.

I'm also leaning on the `Intl.DateTimeFormat` api here to compute the 12-hour time format string (also taking into account the order in which the meridiem appears, either before or after the time value, and whether there's a space in between the meridiem and the time value), since down the road its likely we'll be using `dayjs` much less once we build the masked input experience in `input-time-picker`.